<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    </style>
    <title>Python CGI</title>
    <link rel="stylesheet" href="./css/styles.css">
</head>
<!DOCTYPE html>
<html>
  <body>
    <h4 id="name">THAIS NISENBAUM</h4>
    <div class="cgiEx">
        <h1>Python Computer Gateway Interface (CGI) Programming</h1>
        <hr>
        <h2>HTML Template</h2>
        <h3>GenBank Reference Annotation vs. Glimmer3</h3>
        <p id="justify">Glimmer was used to predict the genes of a plasmid of the model organism <i>Escherichia coli</i> O157:H7 str. Sakai plasmid pO157 DNA, and the results were then compared with the annotations from GenBank. A CGI script was used to read both the GenBank (reference) annotation and predicted output from Glimmer, and the results were populated in an HTML5-compliant template.</p>
            <p id="justify">The output includes summary information at the top, including the count of genes in the reference annotation, the count of predicted genes, the count of genes with exact matching coordinates between reference and prediction (5&#39; and 3&#39; agreement), the count of genes with 5&#39; agreement but 3&#39; disagreement, the count of genes with 5&#39; disagreement but 3&#39; agreement, and the count of genes predicted with no overlap to a reference gene. Additionally, a table is displayed showing all gene coordinates of the reference annotation and the corresponding (overlapping) predicted coordinates. Attributes of &#39;agrees&#39; or &#39;disagrees&#39; are applied when compared to the reference coordinates, and a consensus field is generated.</p>

            <i>Escherichia coli</i> O157:H7 str. Sakai plasmid pO157 DNA, complete sequence: <a href="https://www.ncbi.nlm.nih.gov/nuccore/AB011549.2">AB011549.2</a>

            <img src="imgs/cgi_img1.png">

            <p id="justify">Glimmer (Gene Locator and Interpolated Markov ModelER) is a Unix-based program used to locate bacterial genes with 99% sensitivity (1). Glimmer identifies coding sequences using interpolated Markov models (IMMs), which make predictions based on a linear combination of probabilities from oligomers of varying lengths. In general, longer oligomers have higher predictive value if sufficient data is available. Glimmer uses Markov models from the 1st through 8th orders and assesses which order to use based on oligomer frequency, giving a higher weight to frequently occurring oligomers. Glimmer uses longer oligomers when possible and creates IMMs for the possible reading frames, which are then used to score orfs and determine coding sequences. Glimmer&#39;s use of IMM is advantageous over other programs, such as GeneMark, which uses a fixed-order Markov model. GenMark uses a 5th-order Markov model, predicting nucleotides based on the five previous ones, and may result in unreliable predictions if the 5mer does not frequently occur (2).</p>

                <p>The Glimmer3 system was downloaded, and the file was decompressed using the following command: tar xzf glimmer302b.tar.gz</p>

                <p>Running the command above creates a directory called <strong>glimmer3.02</strong></p>

                <img src="imgs/cgi_img2.png">
                The <strong>glim302notes.pdf </strong>file contains the documentation for running Glimmer.

                <p><strong>Commands used to run glimmer3</strong></p>
           <pre>
                    <code>long-orfs -n -t 1.15 AB011549_seq.fasta AB011549_seq.longorfs
                    </code>
                </pre>
                    <img src="imgs/cgi_img3.png">
                    <br/><br/>
                    <pre>
                    <code>extract -t AB011549_seq.fasta AB011549_seq.longorfs &#62; AB011549_seq.train</code>
                    <br/>
                    <code>build-icm -r AB011549_seq.icm &#60; AB011549_seq.train</code>
                    <br/>
                    <code>glimmer3 -o50 -g110 -t30 AB011549_seq.fasta AB011549_seq.icm AB011549_seq</code>
                    </pre>
                    <br/>
                    <img src="imgs/cgi_img4.png">
                    <br/>
                    <br/>

                    <pre>
                        <code>extract -t AB011549_seq.fasta AB011549_seq.predict &#62; AB011549_seq.glimmer</code>
                    </pre>

                    <h4>Explanation</h4>
                    <p><strong>Command 1: </strong><code>long-orfs -n -t 1.15 AB011549_seq.fasta AB011549_seq.longorfs</code></p>

                    <p>The <strong>long-orfs </strong>command reads a DNA sequence from a file and provides the coordinates of the long, non-overlapping open reading frames using an amino-acid distribution model. The output can be saved to a separate file or printed to the console. The orfs in the longorfs file are likely to contain genes and can be used as a training set for the build-icm program. The <strong>AB011549_seq.fasta </strong>was used for this prediction.</p>

                    <p>It is used as <code>long-orfs [options] &#60;sequence-file&#62; &#60;output-file&#62;</code></p>
                    <em>(“-” instead of &#60;output-file&#62; for standard output)</em>

                    <p><u>Parameters:</u><br/>
                       <strong>-n (--no_header):</strong> Does not include headers in the output <br/> 
                       <strong>-t 1.15 (--cutoff &#60;x&#62;):</strong> Only genes with entropy distance score less than 1.15 will be considered
                    </p>

                    <p>Output (first ten rows; a total of 42 orfs were identified): <strong>AB011549_seq.longorfs</strong>
                        <img src="imgs/cgi_img5.png">
                    </p>

                    <p><strong>Command 2: </strong><code>extract -t AB011549_seq.fasta AB011549_seq.longorfs &#62; AB011549_seq.train</code></p>

                    <p>The <strong>extract </strong>program reads a DNA sequence from a file and takes coordinates to output a file containing FASTA sequences of the specified regions.</p>

                    <p>It is used as <code>extract [options] &#60;sequence-file&#62; &#60;coords&#62;</code></p>

                    <p><u>Parameters:</u><br/>
                       <strong>-t (--nostop):</strong> Omit the last 3 characters of each output string<br/> 
                    </p>

                    <p>Output: <strong>AB011549_seq.train</strong>
                        <img src="imgs/cgi_img6.png">
                    </p>

                    <p><strong>Command 3: </strong><code>build-icm -r AB011549_seq.icm &#60; AB011549_seq.train</code></p>

                    <p>The <strong> build-icm </strong> program constructs a probability model called interpolated context model (ICM) from a set of sequences.</p>

                    <p>It is used as <code>build-icm [options] output_file &#60; input_file</code></p>

                    <p><u>Parameters:</u><br/>
                       <strong>-r (--reverse):</strong> Use the reverse of input strings to build the ICM model.<br/> 
                    </p>

                    <p>Output: <strong>AB011549_seq.icm</strong>
                        <img src="imgs/cgi_img7.png">
                    </p>

                    <p><b>Command 4: </b><code>glimmer3 -o50 -g110 -t30 AB011549_seq.fasta AB011549_seq.icm AB011549_seq</code></p>

                    <p>The <b>glimmer3 </b>program makes gene predictions based on DNA sequences and the ICM model.</p>

                    <p>It is used as <code>glimmer3 [options] &#60;sequence-file&#62; &#60;icm-file&#62; &#60;tag&#62;</code></p>
                    <em>&#60;tag&#62; is used to name two output files: <b>tag.detail</b> and <b>tag.predict</b></em>

                    <p><u>Parameters:</u><br/>
                        <b>-o50 (--max_olap &#60;n&#62;):</b> Set maximum overlap length to 50.<br/> 
                        <b>-g110 (--gene_len &#60;n&#62;):</b> Set minimum gene length to 110.<br/> 
                        <b>-t30 (--threshold &#60;n&#62;):</b> Set threshold score for calling as gene to 30.<br/> 
                     </p>

                     <p>Output (first ten rows): <b>AB011549_seq.predict</b>
                        <img src="imgs/cgi_img8.png">
                    </p>

                    <p>Output: <b>AB011549_seq.detail</b></p>

                    <p><strong>Command 5: </strong><code>extract -t AB011549_seq.fasta AB011549_seq.predict &#62; AB011549_seq.glimmer</code></p>

                    <p>Output: <b>AB011549_seq.glimmer</b></p>

                    <h4>Summary</h4>

                    <p id="justify">The GenBank annotation for the Escherichia coli O157:H7 str. Sakai plasmid pO157 contained records for 83 genes, 2 of which were not provided CDS locations. Glimmer3 predicted 107 genes. Overall, Glimmer3 predicted 51 genes matching both 5&#39;UTR and 3&#39;UTR locations and 23 genes matching only the 3&#39;UTR location as annotated by GenBank. There were no predictions matching only the 5&#39;UTR location.</p>
                    <p id="justify">A total of 9 genes did not match Glimmer3 predictions. Given CDS locations were not provided for two genes in the GenBank annotation, 92% of GenBank genes had at least one matching region with the genes predicted by Glimmer3. In addition to matching CDS locations, 3 GenBank genes also had overlapping regions with genes predicted by Glimmer. Summary tables of the 3 reference genes with matches and overlapping regions are included below.</p>
                    <p id="justify">A total of 33 additional genes were predicted by Glimmer3 that did not match the GenBank reference annotation. Only 69% of the genes predicted by Glimmer3 had at least one matching location to the reference annotation. Given 3 had overlapping regions with reference genes, 72% of all genes predicted by Glimmer had at least one matching region or overlapping region.</p>

                    <p>References:</p>
                    <ol>
                        <li>Delcher, A. L., Bratke, K. A., Powers, E. C., & Salzberg, S. L. (2007). Identifying bacterial genes and endosymbiont DNA with Glimmer. Bioinformatics (Oxford, England), 23(6), 673&#150;679.<a href="https://doi.org/10.1093/bioinformatics/btm009">https://doi.org/10.1093/bioinformatics/btm009</a></li>
                        <li>Salzberg, S. L., Delcher, A. L., Kasif, S., & White, O. (1998). Microbial gene identification using interpolated Markov models. Nucleic acids research, 26(2), 544&#150;548.<a href="https://doi.org/10.1093/nar/26.2.544">https://doi.org/10.1093/nar/26.2.544</a></li>
                        <li>Delcher, A. L. (2006, May 9). Glimmer Release Notes Version 3.02. Retrieved from <a href="https://ccb.jhu.edu/software/glimmer/glim302notes.pdf">https://ccb.jhu.edu/software/glimmer/glim302notes.pdf</a></li>

                    </ol>


                    <h2>MySQL, HTML Template</h2>
                    <h3>Gene Product Search</h3>

                    <p id="justify">Using Python CGI programming, I&#39;ve connected to a SQL database to query results based on user input from an HTML form. The HTML form allows users to enter a search term, which will be used to query the database in the CGI script.</p>

                    CGI script:
                    <img src="imgs/cgi_img9.png">
                    <br/><br/>
                    <img src="imgs/cgi_img10.png">
                    <img src="imgs/cgi_img11.png">


                    <h2>MySQL, Javascript, and JQuery</h2>
                    <h3>Gene Product Search</h3>
                    <p id="justify">The script above was replicated using Javascript and JQuery for client-side interaction, implementing the JQuery autocomplete functionality. The CGI script processes the results and returns them in a JSON structure, which is parsed by Javascript.</p>

                    <img src="imgs/cgi_img12.png">
                    <img src="imgs/cgi_img13.png">

                    <h3>Analysis of Genes Associated with Cancer: Variant Association and Drug Interactions</h3>

                    <p>This tool utilizes two publicly available datasets: <br/><br/>
                    <a href="https://www.disgenet.org/"> DIsGeNET:</a> A publicly available database of genes and variants associated with human diseases <br/>
                    <a href="https://www.dgidb.org/"> Drug-Gene Interaction Database (DGIdb):</a> a drug-gene interaction database open-source tool to generate data on genes, variants, disease attributes, and drug-gene interactions </p>

                    <p>The home page of the tool allows users to choose between searching for <b>Variant Associations</b> or <b>Drug Interactions</b>.</p>

                    <img src="imgs/cgi_img14.png">

                    <h4>Variant Associations</h4>
                   <p>Enter a gene name and select one of the six cancer types:<i>Breast Carcinoma, Colorectal Carcinoma, Liver Carcinoma, Non-Small Cell Lung Carcinoma, Prostate Carcinoma, and Renal Cell Carcinoma</i></p>

                    <img src="imgs/cgi_img15.png">
                    <img src="imgs/cgi_img16.png">
                    <img src="imgs/cgi_img17.png">

                    <h4>Drug Interactions</h4>
                    <p>Enter a gene name to obtain gene-drug interactions from three sources: Cancer Commons, My Cancer Genome, and My Cancer Genome Clinical Trial</p>

                    <img src="imgs/cgi_img18.png">
                    <img src="imgs/cgi_img19.png">

                    <h4>Tool Description</h4>
                    <ol>
                        <li><b>SQL Relational Database - </b>Stores information about genes and variants associated with cancer and drug-gene interactions. Information about the former was obtained from the DisGeNET SQLite file, which consists of eight tables containing information about genes, variants, and disease attributes and how they relate to one another. Table 1, as referenced above, consists of a combination of data from various tables. Information about drug-gene interactions was obtained from raw data files in DGIdb. </li>
                        <li><b>Python Computer Gateway Interface (CGI) Programming - </b>Connects to the database and queries the data based on user input from an HTML form. The CGI script will process and store the results in the two tables referenced above. The results will be returned in a JSON structure, which will be displayed to the user.</li>
                        <li><b>Javascript and JQuery - </b>The JSON data structure will be read and parsed by Javascript, and the resulting tables will be generated to be displayed to the user. The resulting page will be styled with CSS.</li>
                    </ol>








 </div>

</body>
</html>