<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    </style>
    <title>Bioinformatics Portfolio</title>
    <link rel="stylesheet" href="styles.css">
  </head>
<!DOCTYPE html>
<html>
  <body>
    <h1>BIOINFORMATICS PORTFOLIO</h1>

    <h2>Introduction</h2>

    <p>
      Hello, and welcome to my bioinformatics portfolio. I am a highly motivated
      bioinformatics professional with a passion for solving complex biological
      problems using computational methods.
    </p>
    <p>
      I am currently completing my masterâ€™s degree in Biotechnology with a
      concentration in Bioinformatics from Johns Hopkins University. Throughout
      my studies, I have gained experience in various areas of bioinformatics,
      including genomics, transcriptomics, and proteomics.
    </p>
    <p>
      My portfolio highlights my skills and expertise in bioinformatics,
      featuring projects and code snippets that demonstrate my ability to
      analyze, interpret, and visualize biological data. In addition, my
      portfolio includes examples of my proficiency in software development, as
      I have designed and implemented various bioinformatics tools and pipelines
      using programming languages such as Python, R, and SQL. I have also worked
      on several projects that involve the integration of multiple data types,
      including RNA-seq and ChIP-seq, and am experienced in using various
      bioinformatics databases and tools, including NCBI, Ensembl, and UCSC
      Genome Browser.
    </p>
    <p>
      Overall, my bioinformatics portfolio is a testament to my dedication and
      passion for bioinformatics, demonstrating my eagerness to continue
      developing my skills and knowledge in the field. I am excited about the
      opportunities to collaborate with others in the field and contribute to
      advancing our understanding of the complex biological systems around us.
    </p>
    <p>Thank you for taking the time to review my portfolio.</p>

    <h2>Skills Summary</h2>

    <p>
      <strong>Programming Languages</strong>
    </p>
    <p>Python, SQL, R, Javascript/JQuery</p>
    <p>
      <strong>R/Bioconductor Packages</strong>
    </p>
    <p>
      ggplot2, peptides, biomaRt, biostrings, seqinr, cn.mops, CNVtools, edgeR
    </p>
    <p>
      <strong>Prokaryotic Gene Prediction and Annotation</strong>
    </p>
    <p>
      Gene-Finding Programs: ORF Finder, Glimmer, FGENESB, GeneMark, EasyGene
      1.2b, Prodigal
    </p>
    <p>Promoter-Finding Programs: BROM, Neural Network Promoter Prediction</p>
    <p>
      <strong>Eukaryotic Gene Prediction</strong>
    </p>
    <p>
      Single Genome De Novo Gene Prediction: GENSCAN, HMMGene, FGENESH, Augustus
    </p>
    <p>Expression-Based Prediction: Splign, Genomic BLAST, BLAT</p>
    <p>
      <strong>Protein Prediction</strong>
    </p>
    <p>
      PSORTb, Hierarchical Neural Network (HNN), PHD, Protein Homology/analogY
      Recognition Engine V 2.0 (PHYRE2)
    </p>
    <p>
      <strong>Functional RNAs</strong>
    </p>
    <p>PicTar, PPfold</p>
    <p>
      <strong>Genomic Databases and Tools</strong>
    </p>
    <p>
      NCBI, Genome Data Viewer (GDV), NCBI Variation Viewer, Ensembl, UCSC
      (Genome Browser, Table Browser), Galaxy, IGV, dbSNP, ClinVar, SNPEffect
      and SNPedia, Database of Genomic Variants (DGV), Genome Aggregation
      Database (GnomAD), InterPro, ENCODE, NONCODE, TargetScan, miRNEST 2.0,
      PolymiRTS, DIANA-TarBase v.8, miRDB, miRBase, Rfam, cneViewer, UCNEbase
    </p>
    <p>
      <strong>Sequence Alignment </strong>
    </p>
    <p>MUSCLE, Clustal Omega, MAFFT</p>
    <p>
      <strong>Next Generation Sequencing Analysis</strong>
    </p>
    <p>
      Velvet, BWA, Bowtie2, HISAT, FreeBayes, MACS2, Stringtie, DESeq2,
      BEDtools, SAMtools, GATK, SnpEff, ANNOVAR, TopHat, Cufflinks
    </p>
    <h1></h1>
  </body>
</html>
