<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Bioinformatics Portfolio</title>
    <link rel="stylesheet" href="./css/index.css">
    <link
    rel="stylesheet"
    href="//code.jquery.com/ui/1.13.2/themes/base/jquery-ui.css"
  />
  <script
    type="text/javascript"
    src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"
  ></script>
  <script type="text/javascript" src="js/pages.js"></script>
  <script src="https://code.jquery.com/ui/1.13.2/jquery-ui.js"></script>
</head>
<html>
  <body>
    <h4>THAIS NISENBAUM</h4>

    <h1>BIOINFORMATICS PORTFOLIO</h1> 

      <div class="main_tabs">
        <ul class="tabs">
          <li id="resume_tab">
            <a href="resume.html">Resume</a>
          </li>
          <li id="samples_tab">
            <a href="samples.html">Work Samples</a>
          </li>
        </ul>
      </div>

      <hr>

      <div id="intro">

        <h2>INTRODUCTION</h2>

        <p>
            Hello, and welcome to my Bioinformatics portfolio. I am a highly motivated
            bioinformatics professional with a passion for solving complex biological
            problems using computational methods.I am currently completing my masterâ€™s degree in Biotechnology with a concentration in Bioinformatics from Johns Hopkins University. Throughout
            my studies, I have gained experience in various areas of bioinformatics,
            including genomics, transcriptomics, and proteomics. My thesis involves exploring the potential of DNA-Based digital twin technology for the advancement of precision medicine in tongue cancer.
          </p>
          <p>
            My portfolio highlights my skills and expertise in bioinformatics,
            featuring projects and code snippets that demonstrate my ability to
            analyze, interpret, and visualize biological data. In addition, my
            portfolio includes examples of my proficiency in software development, as
            I have designed and implemented various bioinformatics tools and pipelines
            using programming languages such as Python, R, and SQL. I have also worked
            on several projects that involve the integration of multiple data types,
            including RNA-seq and ChIP-seq, and am experienced in using various
            bioinformatics databases and tools, including NCBI, Ensembl, and UCSC
            Genome Browser.
          </p>
          <p>
            Overall, my bioinformatics portfolio is a testament to my dedication and
            passion for bioinformatics, demonstrating my eagerness to continue
            developing my skills and knowledge in the field. I am excited about the
            opportunities to collaborate with others in the field and contribute to
            advancing our understanding of the complex biological systems around us.
          </p>
          <p>Thank you for taking the time to review my portfolio.</p>

          </div>

          <div id="summary">

          <h2>SKILLS SUMMARY</h2>

    <hr>


    <h3>Programming Languages</h3>
    <p>
        Python, SQL, R, Javascript/JQuery
    </p>

    <h3>R/Bioconductor Packages</h3>
    <p>
        ggplot2, peptides, biomaRt, biostrings, seqinr, cn.mops, CNVtools, edgeR
    </p>

    <h3>Prokaryotic Gene Prediction and Annotation</h3>
    <p> Gene-Finding Programs: ORF Finder, Glimmer, FGENESB, GeneMark, EasyGene
        1.2b, Prodigal</p>
        <p>Promoter-Finding Programs: BROM, Neural Network Promoter Prediction
        </p>
      <h3>Eukaryotic Gene Prediction</h3>
    </p>
    <p>
      Single Genome De Novo Gene Prediction: GENSCAN, HMMGene, FGENESH, Augustus
    </p>
    <p>Expression-Based Prediction: Splign, Genomic BLAST, BLAT</p>
    <p>
      <h3>Protein Prediction</h3>
    </p>
    <p>
      PSORTb, Hierarchical Neural Network (HNN), PHD, Protein Homology/analogY
      Recognition Engine V 2.0 (PHYRE2)
    </p>
    <p>
      <h3>Functional RNAs</h3>
    </p>
    <p>PicTar, PPfold</p>
    <p>
      <h3>Genomic Databases and Tools</h3>
    </p>
    <p>
      NCBI, Genome Data Viewer (GDV), NCBI Variation Viewer, Ensembl, UCSC
      (Genome Browser, Table Browser), Galaxy, IGV, dbSNP, ClinVar, SNPEffect
      and SNPedia, Database of Genomic Variants (DGV), Genome Aggregation
      Database (GnomAD), InterPro, ENCODE, NONCODE, TargetScan, miRNEST 2.0,
      PolymiRTS, DIANA-TarBase v.8, miRDB, miRBase, Rfam, cneViewer, UCNEbase
    </p>
    <p>
      <h3>Sequence Alignment </h3>
    </p>
    <p>MUSCLE, Clustal Omega, MAFFT</p>
    <p>
      <h3>Next Generation Sequencing Analysis</h3>
    </p>
    <p>
      Velvet, BWA, Bowtie2, HISAT, FreeBayes, MACS2, Stringtie, DESeq2,
      BEDtools, SAMtools, GATK, SnpEff, ANNOVAR, TopHat, Cufflinks
    </p>


      

          </div>
    
         

  </body>
</html>
