<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    </style>
    <title>R</title>
    <link rel="stylesheet" href="./css/styles.css">
</head>
<!DOCTYPE html>
<html>
  <body>
    <h4>THAIS NISENBAUM</h4>
    <div class="rEx">

    <h1>R</h1>
    <hr>
    <h2>Data Analysis and Visualization</h2>
    <p><strong>Example 1:</strong>Assay of serum free light chain for 7874 subjects</p>

    <p>Analysis of the <i>flchain</i> dataset from the<a href="https://cran.r-project.org/web/packages/survival/survival.pdf">R survival package</a></p>

    <img src="imgs/rex1_img1.png" width="580" height="450">
    <img src="imgs/rex1_img2.png" width="580" height="300">
    <img src="imgs/rex1_img3.png" width="580" height="400">
    <img src="imgs/rex1_img4.png" width="580" height="400">
    <img src="imgs/rex1_img5.png" width="580" height="400">
    <img src="imgs/rex1_img6.png" width="580" height="400">
    <img src="imgs/rex1_img7.png" width="580" height="400">
    <img src="imgs/rex1_img8.png" width="580" height="400">
    <img src="imgs/rex1_img9.png" width="580" height="400">
    <img src="imgs/rex1_img10.png" width="580" height="400">
    <img src="imgs/rex1_img11.png" width="580" height="400">
    <img src="imgs/rex1_img12.png" width="580" height="400">

    <p><strong>Example 2:</strong>Ten Most Commonly Mutated Genes in Six Cancer Types</p>
    <p>Output used in the Analysis of Genes Associated with Cancer: Variant Association and Drug Interactions referenced under Python Computer Gateway Interface (CGI) Programming</p>

    <img src="imgs/rex2_img1.png" width="580" height="300">
    <img src="imgs/rex2_img2.png" width="580" height="450">
    <img src="imgs/rex2_img3.png" width="580" height="1000">
    <img src="imgs/rex2_img4.png" width="580" height="1000">


    <h2>Data Mining </h2>
    <h3>Ensembl BiomaRt</h3>
    <p><strong>Ensembl </strong>is a genome browser for chordate sequenced genomes and non-chordate model organisms. The data from Ensembl is stored in MySQL relational databases and can be retrieved using BioMart, which contains several categories in which a query can be filtered. <strong>BioMart</strong> can be accessed through the web or through the Bioconductor package BiomaRt. Bioconductor is an open-source tool to mine, analyze, and visualize data</p>

    <a href="http://useast.ensembl.org/biomart/martview/f1ac5790705209ca374707ee4d91c0b3">Ensembl - BioMart</a><br/>
    <a href="https://bioconductor.org/packages/release/bioc/vignettes/biomaRt/inst/doc/biomaRt.html">Bioconductor - BiomaRt</a>

    <p><strong>Example 1:</strong>According to information from OMIM.org for Huntington's disease, biomaRt was used to retrieve two tables containing the following attributes for five MIM identifiers.</p>

    <p>Table 1: Entrez Gene ID, HGNC symbol, Ensembl Gene ID</p>
    <p>   Table 2: HGNC symbol, Ensembl Gene ID, Ensembl Transcript ID</p>
    <p> Table 1 is gene-specific, with each line representing one gene. Table 2 has more lines because it shows transcripts, and each gene can have more than one transcript.</p>

    <img src="imgs/biomart_img1.png" width="580" height="250">
    <img src="imgs/biomart_img2.png" width="580" height="200">
    <img src="imgs/biomart_img3.png" width="580" height="900">

    <h2>Sequence Analysis and Manipulation</h2>
    <h3>Seqnir</h3>
<p><strong>Seqnir</strong> is an R package used to analyze DNA and protein sequences from databases (FASTA format).</p>
<p>More information about Seqnir can be found<a href="https://a-little-book-of-r-for-bioinformatics.readthedocs.io/en/latest/src/chapter1.html">here</a></p>

<p><strong>Example:</strong>Analysis of <i>Pseudomonas alkylphenolia</i>train KL28 genome sequence using the Seqnir R package</p>

<img src="imgs/seqnir_img1.png" width="580" height="500">


</div>

  
</body>
</html>